Core/src/org/sleuthkit/autopsy/geolocation/datamodel/Track.java

```diff
@@ -130,14 +133,17 @@ private void buildPath(GeoTrackPoints points, BlackboardArtifact artifact) throw
      */
     private GeoTrackPoints getPointsList(Map<BlackboardAttribute.ATTRIBUTE_TYPE, BlackboardAttribute> attributeMap) throws GeoLocationDataException {
         BlackboardAttribute attribute = attributeMap.get(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_TRACKPOINTS);
-        if (attribute != null) {
-            try {
-                return BlackboardJsonAttrUtil.fromAttribute(attribute, GeoTrackPoints.class);
-            } catch (InvalidJsonException ex) {
-                throw new GeoLocationDataException("Unable to parse track points in TSK_GEO_TRACKPOINTS attribute", ex);
-            }
+        if (attribute == null) {
```

Comment:
Can this become an illegal argument? And preferably out in the constructor of Track? I'm worried that GeoLocationDataException is becoming a 'god' exception.

Reply:
I had a conversation where I think we decided on a `GeoLocationDataException`, because, similar to json parsing, the artifact attributes being passed were in a malformed state if they did not contain track information.